test/python/test_nexus_multipath.py

```diff
@@ -336,3 +409,71 @@ def test_nexus_multipath_add_3rd_path(
 
     # wait for fio to complete
     time.sleep(15)
+
+
+@pytest.mark.timeout(60)
+def test_nexus_multipath_remove_3rd_path(
+    create_nexus_dev,
+    create_nexus_2_no_destroy,
+    connect_nexus_2,
+    create_nexus_3_dev,
+    start_fio,
+    destroy_nexus_2,
+):
+    """Create 3 nexuses, connect over NVMe, start fio, destroy 2nd nexus."""
+
+    dev = create_nexus_dev
+    dev2 = connect_nexus_2
+    start_fio
+    dev3 = create_nexus_3_dev
+    assert dev == dev2, "should have one namespace"
+    assert dev == dev3, "should have one namespace"
+
+    desc = nvme_list_subsystems(dev)
+    paths = desc["Subsystems"][0]["Paths"]
+    assert len(paths) == 3, "should have 3 paths"
+
+    assert paths[0]["State"] == "live"
+    # kernel 5.4 reports resetting, 5.10 reports connecting
+    assert paths[1]["State"] == "resetting" or paths[1]["State"] == "connecting"
+    assert paths[2]["State"] == "live"
+
+    # wait for fio to complete
+    time.sleep(15)
+
+
+@pytest.mark.timeout(60)
+def test_nexus_multipath_remove_all_paths(
+    create_nexus_no_destroy,
+    create_nexus_2_no_destroy,
+    connect_nexus,
+    connect_nexus_2,
+    start_fio,
+    unpublish_nexus,
+    delay,
+    verify_paths,
+    destroy_nexus_2,
+    delay2,
+    publish_nexus,
+):
+    """Create 2 nexuses, connect over NVMe, start fio, unpublish one nexus,
+    verify failover, destroy the other nexus, re-publish the first nexus,
+    verify IO restarts."""
+
+    dev = connect_nexus
+    dev2 = connect_nexus_2
+    start_fio
+    assert dev == dev2, "should have one namespace"
+
+    # wait for reconnection
+    time.sleep(10)
+
+    desc = nvme_list_subsystems(dev)
+    paths = desc["Subsystems"][0]["Paths"]
+    assert len(paths) == 2, "should have 2 paths"
+
+    assert paths[0]["State"] == "live"
+    assert paths[1]["State"] == "connecting"
```

Comment:
Does it eventually become "live"? If so, should we test for it?

Reply:
In this test, no, as the second nexus remains destroyed and is not recreated.